handlers/download.go

```diff
@@ -147,3 +154,32 @@ func (d Download) Do(extension string) http.HandlerFunc {
 		http.Error(w, notFoundMessage, http.StatusNotFound)
 	}
 }
+
+func (d Download) authenticate(r *http.Request, logData map[string]interface{}) (bool, map[string]interface{}) {
+	var authorised bool
+
+	if d.IsPublishing {
+		var hasCallerIdentity, hasUserIdentity bool
+
+		callerIdentity := identity.Caller(r.Context())
+		if callerIdentity != "" {
```

Comment:
If you decide not to use the `identity.isPresent(ctx)` function - consider also checking for a nil value here